inference-engine/src/mkldnn_plugin/nodes/mkldnn_shuffle_channels_node.cpp

```diff
@@ -0,0 +1,338 @@
+                    optimizedParams.src_block_dims[i - 1] = dataDims[i];
+                }
+            }
+            optimizedParams.src_block_dims[batchRank - 1] = group;
+            optimizedParams.src_block_dims[batchRank] = groupSize;
+            order[batchRank - 1] = batchRank;
+            order[batchRank] = batchRank - 1;
+
+            if (sptialRank != 0) {
+                order[batchRank + 1] = batchRank + 1;
+                optimizedParams.src_block_dims[batchRank + 1] = spatialShapeSize;
+            }
+        } else { // axis on batch
+            order[0] = 1;
+            optimizedParams.src_block_dims[0] = group;
+            order[1] = 0;
+            optimizedParams.src_block_dims[1] = groupSize;
+            order[2] = 2;
+            optimizedParams.src_block_dims[2] = spatialShapeSize;
+        }
+    } else {
+        for (int i = 0; i < batchRank; i++) {
+            optimizedParams.src_block_dims[i] = dataDims[i];
+            order[i] = i;
+        }
+
+        optimizedParams.src_block_dims[batchRank] = group;
+        optimizedParams.src_block_dims[batchRank + 1] = groupSize;
+        order[batchRank] = batchRank + 1;
+        order[batchRank + 1] = batchRank;
+
+        if (sptialRank != 0) {
+            order[batchRank + 2] = batchRank + 2;
+            optimizedParams.src_block_dims[batchRank + 2] = spatialShapeSize;
+        }
+    }
+
+    prepareOptimizedParams(getSelectedPrimitiveDescriptor()->getConfig().inConfs[0].desc.getPrecision());
+    prepareConfigParams();
+}
+
+void MKLDNNShuffleChannelsNode::execute(mkldnn::stream strm) {
+    if (permute_kernel) {
+        auto srcData = reinterpret_cast<const uint8_t*>(this->getParentEdgeAt(0)->getMemoryPtr()->GetPtr());
+        auto dstData = reinterpret_cast<uint8_t*>(this->getChildEdgeAt(0)->getMemoryPtr()->GetPtr());
+        optimizedExecute(srcData, dstData);
+    } else {
+        auto srcData = reinterpret_cast<const float*>(this->getParentEdgeAt(0)->getMemoryPtr()->GetPtr());
+        auto dstData = reinterpret_cast<float*>(this->getChildEdgeAt(0)->getMemoryPtr()->GetPtr());
+        executeRef(srcData, dstData);
+    }
+}
+
+void MKLDNNShuffleChannelsNode::executeRef(const float* srcData, float* dstData) {
+    size_t dataLength = 1;
+    size_t workAmountDst;
+    size_t ownDims[CNTR_SIZE];
+    size_t ownStrides[CNTR_SIZE];
+
+    ownDims[0] = 1;
+    for (int i = 0; i < axis; i++)
+        ownDims[0] *= dataDims[i];
+
+    for (size_t i = axis + 1; i < dataDims.size(); i++)
+        dataLength *= dataDims[i];
+
+    if (dataLength == 0)
+        THROW_SHCH_ERROR << "has incorrect input parameters dimension.";
+
+    ownDims[1] = groupSize;
+    ownDims[2] = group;
+    ownStrides[0] = dataDims[axis];
+    ownStrides[1] = 1;
+    ownStrides[2] = ownDims[1];
+    workAmountDst = ownStrides[0] * ownDims[0];
+
+    auto initter = [&](size_t start, size_t size, size_t* counters, size_t* ownDims, size_t* ownStrides) -> size_t {
```

Comment:
If variables `size_t* ownDims, size_t* ownStrides` are constant and before using this function we know these variables, may be should remove it from the lambda function parameters?
The same comment for `updater`

Reply:
you are right. fixed, thank you!